base/libc.jl

```diff
@@ -82,8 +85,16 @@ flush_cstdio() = ccall(:jl_flush_cstdio, Void, ())
 ## time-related functions ##
 
 # TODO: check for usleep errors?
-@unix_only systemsleep(s::Real) = ccall(:usleep, Int32, (UInt32,), round(UInt32,s*1e6))
-@windows_only systemsleep(s::Real) = (ccall(:Sleep, stdcall, Void, (UInt32,), round(UInt32,s*1e3)); return Int32(0))
+if is_unix()
+    systemsleep(s::Real) = ccall(:usleep, Int32, (UInt32,), round(UInt32, s*1e6))
+elseif is_windows()
+    function systemsleep(s::Real)
+        ccall(:Sleep, stdcall, Void, (UInt32,), round(UInt32, s * 1e3))
+        return Int32(0)
+    end
+else
+    error("systemsleep undefined for this OS")
```

Comment:
no, although sometimes we instead define the method as throwing an error instead of throwing it immediately